includes/Ajax.php

```diff
@@ -302,6 +302,12 @@ public function change_order_status() {
         $order = dokan()->order->get( $order_id );
         $order->update_status( $order_status );
 
+        // Get the new order status. This is needed since plugin/theme authors might
```

Comment:
@rmilesson why we need to get order status manually? there is a hook to add order status, maybe you could use that?
`apply_filters( 'wc_order_statuses', $order_statuses )`

Reply:
This is not possible in this case since `$order_status` (fetched on line 313) can be out of date after running `$order->update_status(  $order_status ) ` on line 320.
This is why we need to re-fetch `$order_status`.

Reply:
Consider this hook in a theme:

```
add_action(
	'woocommerce_order_status_changed',
	function ( $order_id ) {
		$order = wc_get_order( $order_id );
		
		$order->update_status( 'wc-processing' );
		$order->save();
	}
);
```

This hook would demonstrate the issue by always setting the order status to `wc-processing`. However you would show the wrong order status after updating the order status on the order dashboard.